src/accelerate/inference.py

```diff
@@ -59,9 +62,42 @@ def build_pipeline(model, split_points, args, kwargs) -> PipelineStage:
 def pippy_forward(forward, *args, **kwargs):
     state = PartialState()
     output = None
+
+    def _find_batch_size(arg):
+        try:
+            return find_batch_size(arg)
+        except (ValueError, TypeError):
+            pass
+        return None
+
+    def _pad_inputs(args):
+        # Slice and copy the last input
+        extra = slice_tensors(
+            args,
+            slice(state.num_processes, state.num_processes + 1),
+        )
```

Comment:
Are we use that the last input will be in the index state.num_processes ? Maybe it is better to duplicate the first input ?

Reply:
We do first input in the `DataLoader` class so would be good to do the same here